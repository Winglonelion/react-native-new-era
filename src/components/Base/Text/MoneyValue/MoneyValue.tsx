import React, { memo } from 'react';
import { Text } from 'react-native';

interface PropTypes {
  value: number;
  fixed?: number;
}

const MoneyValue: React.FC<PropTypes> = memo(({ value, fixed = 2 }) => {
  return <Text>{`$${value.toFixed(fixed)}`}</Text>;
});

export default MoneyValue;
