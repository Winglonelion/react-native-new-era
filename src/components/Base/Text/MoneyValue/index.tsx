export { default as default } from './MoneyValue';
