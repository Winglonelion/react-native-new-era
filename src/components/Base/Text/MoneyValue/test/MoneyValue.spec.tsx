import React from 'react';
import { shallow } from 'enzyme';
import MoneyValue, { PropTypes } from '../MoneyValue';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<MoneyValue {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('MoneyValue check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({ value: 90000, fixed: 2 });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot without props', () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
