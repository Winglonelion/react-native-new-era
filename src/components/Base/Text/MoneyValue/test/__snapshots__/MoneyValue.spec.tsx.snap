// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyValue check snapshot should match snapshot with props 1`] = `
<Text>
  $90000.00
</Text>
`;

exports[`MoneyValue check snapshot should match snapshot without props 1`] = `""`;
